core/templates/pages/preferences-page/preferences-page.component.ts

```diff
@@ -176,10 +179,13 @@ angular.module('oppia').component('preferencesPage', {
 
         ctrl.username = '';
         LoaderService.showLoadingScreen('Loading');
-        var userInfoPromise = UserService.getUserInfoAsync();
+        var userInfoPromise = UserBackendApiService.getUserInfoAsync();
         userInfoPromise.then(function(userInfo) {
           ctrl.username = userInfo.getUsername();
           ctrl.email = userInfo.getEmail();
+          // TODO(#8521): Remove the use of $rootScope.$apply()
+          // once the controller is migrated to angular.
+          $rootScope.$apply();
```

Comment:
I think the execution doesn't pass line number 185. So probably `rootScope.apply` is not covered (i.e. would be in red in the coverage report). I could be wrong, so check that once. If my hypothesis is true, i.e. `rootScope.apply` is not covered, then check if the function `getEmail()` exists on userInfo object.

Reply:
Done!